pkg/client/client.go

```diff
@@ -12,7 +12,12 @@ import (
 type SecretsClient interface {
 	SetSecret(ctx context.Context, storeName, id string, request *types.SetSecretRequest) (*types.SecretResponse, error)
 	GetSecret(ctx context.Context, storeName, id, version string) (*types.SecretResponse, error)
+	GetDeletedSecret(ctx context.Context, storeName, id string) (*types.SecretResponse, error)
+	DeleteSecret(ctx context.Context, storeName, id string) error
+	RestoreSecret(ctx context.Context, storeName, id string) error
+	DestroySecret(ctx context.Context, storeName, id string) error
 	ListSecrets(ctx context.Context, storeName string) ([]string, error)
+	ListDeletedSecrets(ctx context.Context, storeName string) ([]string, error)
```

Comment:
`GetDeleted` and `ListDeleted` are just Get and List but with a query param. This is the example for Eth1 in the handler:
```
getDeleted := request.URL.Query().Get("deleted")
	var eth1Acc *entities.ETH1Account
	if getDeleted == "" {
		eth1Acc, err = eth1Store.Get(ctx, getAddress(request))
	} else {
		eth1Acc, err = eth1Store.GetDeleted(ctx, getAddress(request))
	}
```

Reply:
I didn't know we impleted already a pattern for it. I will apply same pattern here